crates/types/src/consensus.rs

```diff
@@ -57,106 +58,201 @@ pub struct Consensus<TYPES: NodeType, LEAF: LeafType<NodeType = TYPES>> {
-    /// Used for metrics.  This resets to 0 on every decide event.
-    pub invalid_qc: usize,
+    pub metrics: Arc<ConsensusMetricsValue>,
 }
 
-/// The metrics being collected for the consensus algorithm
-pub struct ConsensusMetrics {
+/// Contains several `ConsensusMetrics` that we're interested in from the consensus interfaces
+#[derive(Clone, Debug)]
+pub struct ConsensusMetricsValue {
+    #[allow(dead_code)]
+    /// The values that are being tracked
+    pub values: Arc<Mutex<InnerConsensusMetrics>>,
+    /// The number of last synced synced block height
+    pub last_synced_block_height: Box<dyn Gauge>,
+    /// The number of last decided view
+    pub last_decided_view: Box<dyn Gauge>,
     /// The current view
     pub current_view: Box<dyn Gauge>,
-    /// The duration to collect votes in a view (only applies when this insance is the leader)
-    pub vote_validate_duration: Box<dyn Histogram>,
-    /// The duration we waited for txns before building the proposal
-    pub proposal_wait_duration: Box<dyn Histogram>,
-    /// The duration to build the proposal
-    pub proposal_build_duration: Box<dyn Histogram>,
-    /// The duration of each view, in seconds
-    pub view_duration: Box<dyn Histogram>,
-    /// Number of views that are in-flight since the last committed view
-    pub number_of_views_since_last_commit: Box<dyn Gauge>,
+    // The duration to collect votes in a view (only applies when this insance is the leader)
+    // pub vote_validate_duration: Box<dyn Histogram>,
+    // The duration we waited for txns before building the proposal
+    // pub proposal_wait_duration: Box<dyn Histogram>,
+    // The duration to build the proposal
+    // pub proposal_build_duration: Box<dyn Histogram>,
+    // The duration of each view, in seconds
+    // pub view_duration: Box<dyn Histogram>,
+    /// Number of views that are in-flight since the last decided view
+    pub number_of_views_since_last_decide: Box<dyn Gauge>,
     /// Number of views that are in-flight since the last anchor view
     pub number_of_views_per_decide_event: Box<dyn Histogram>,
-    /// Number of invalid QCs between anchors
-    pub invalid_qc_views: Box<dyn Histogram>,
-    /// Number of views that were discarded since from one achor to the next
-    pub discarded_views_per_decide_event: Box<dyn Histogram>,
-    /// Views where no proposal was seen from one anchor to the next
-    pub empty_views_per_decide_event: Box<dyn Histogram>,
-    /// Number of rejected transactions
+    /// Number of invalid QCs we've seen since the last commit.
+    pub invalid_qc: Box<dyn Gauge>,
+    // Number of views that were discarded since from one anchor to the next
+    // pub discarded_views_per_decide_event: Box<dyn Histogram>,
+    // Views where no proposal was seen from one anchor to the next
+    // pub empty_views_per_decide_event: Box<dyn Histogram>,
+    /// Number of rejected transactions, it's more like duplicated transactions in current implementation
```

Comment:
Nit: can we remove these commented-out fields if they won't be used anymore?

Reply:
Removed! Also removed `#[allow(dead_code)]`.